File: solutions/Python/2390.RemovingStarsFromAString.py
class Solution:
    def removeStars(self, s: str) -> str:
        stack = Stack()

        for c in s:
            if c == "*":
                stack.pop()
            else:
                stack.push(c)

        return str(stack)


class Stack:
    def __init__(self) -> None:
        self._stack: list[str] = []

    def push(self, c) -> None:
        self._stack.append(c)

    def pop(self) -> None:
        if not self.isEmpty():
            self._stack.pop()

    def isEmpty(self) -> bool:
        return len(self._stack) == 0

    def __str__(self) -> str:
        return ''.join(self._stack)

    @property
    def stack(self):
        return self._stack

File: solutions/JavaScript/2390.RemovingStarsFromAString.js
function removeStars(s) {
    const stack = new Stack();

    for (const c of s) {
        if (c === "*") {
            stack.pop();
        } else {
            stack.push(c);
        }
    }

    return stack.toString();
}

class Stack {
    constructor() {
        this._stack = [];
    }

    push(c) {
        this._stack.push(c);
    }

    pop() {
        if (!this.isEmpty()) {
            this._stack.pop();
        }
    }

    isEmpty() {
        return this._stack.length === 0;
    }

    toString() {
        return this._stack.join('');
    }
}
